src/auth.ts

```diff
@@ -0,0 +1,50 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+export const M2_DIR = '.m2';
+export const SETTINGS_FILE = 'settings.xml';
+
+export async function configAuthentication(
+  id: string,
+  username: string,
+  password: string
+) {
+  if (id && username && password) {
+    console.log(
+      `creating ${SETTINGS_FILE} with server-id: ${id}, username: ${username}, and a password`
+    );
+    const directory: string = path.join(os.homedir(), M2_DIR);
```

Comment:
Didn't mention this earlier, but the `$GITHUB_WORKSPACE` directory is where we should be creating any new directories or files. Whenever we use `actions/checkout`, any checked out code gets saved to to the directory so we should mimic the same behavior: https://github.com/actions/checkout

This does also brings up the potential scenario of what-if someone already has a .`m2` with the same settings file. I would add an extra try-catch block around `fs.writeFileSync`. There should be a clear error message and a failure if we're unable to create the necessary files. You can use `@actions/core` to do that: https://github.com/actions/toolkit/tree/master/packages/core

Reply:
> Didn't mention this earlier, but the `$GITHUB_WORKSPACE` directory is where we should be creating any new directories or files. Whenever we use `actions/checkout`, any checked out code gets saved to to the directory so we should mimic the same behavior: https://github.com/actions/checkout

I'll give this a try.  The `settings.xml` file needs to live in `~/.m2/` so if `$GITHUB_WORKSPACE` isn't the same as `$HOME` this might not work. 



> This does bring up the potential scenario of what-if someone already has a .`m2` directory with the same settings file if they've used checkout. You do mention in the example in the `README` that running the action again causes an overwrite to happen. I think it would be good to detect this behavior and output some message to inform the user about this overwrite.

Good idea, I'll add something to check for an existing file and log if we're overwriting it.

Reply:
> I'll give this a try. The `settings.xml` file needs to live in `~/.m2/` so if `$GITHUB_WORKSPACE` isn't the same as `$HOME` this might not work.

This writes to the wrong directory I'm seeing in the logs:
```
writing /home/runner/work/fib-apache-maven/fib-apache-maven/.m2/settings.xml
```

And then the authentication fails.  See https://github.com/pied-piper-inc/fib-apache-maven/commit/19b690b1b6c87e9a7aa468dba112e7f546b13a3f/checks?check_suite_id=343056828

Reply:
Did some more investigation and I can override the setting with the command line flag `-s $GITHUB_WORKSPACE/.m2/settings.xml` which makes the full deploy `mvn deploy -s $GITHUB_WORKSPACE/.m2/settings.xml`

Its not as nice as the straight `mvn deploy` but if the `$GITHUB_WORKSPACE` is a better place than `$HOME` for these kind of auth files we can do that instead.  Lemme know what you think.

Reply:
The end goal is to have a simple experience, and an extra `-s` flag complicates the whole scenario which we don't want to do.

Lets keep it in `$HOME`. I would only just add a small note in the `README` and clarify that this is exactly where the `settings.xml` file will be created.

Reply:
Ok, done!